cranelift-codegen/src/legalizer/mod.rs

```diff
@@ -145,6 +147,16 @@ pub fn legalize_function(func: &mut ir::Function, cfg: &mut ControlFlowGraph, is
 
     func.encodings.resize(func.dfg.num_insts());
 
+    {
+        // Split ebb params before trying to legalize instructions, so that the newly introduced
+        // isplit instructions get legalized.
+
+        let mut pos = FuncCursor::new(func);
+        while let Some(ebb) = pos.next_ebb() {
+            split::split_ebb_params(pos.func, cfg, ebb);
+        }
+    }
+
     let mut pos = FuncCursor::new(func);
```

Comment:
Instead of making two of `FuncCursor` can we move lines 155-157 below this one? Or is there some ownership thing that prevents it?

Reply:
Should be do-able.